### PR TITLE
Minor code format improvements for log statements

### PR DESCRIPTION
**Pull request by newtork**
Unfortunately our formatter is unpleasant with the log statements.
Since this is not going to change anytime soon, let's improve it  by reducing line breaks.

### CODE REVIEW

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java [hunk 1]
```diff
@@ -195,12 +194,8 @@ private List<Header> getHeadersFromHeaderProviders( @Nonnull final URI requestUr
...
         aggregatedHeaderProviders.addAll(headerProvidersFromClassLoading);
 
-        log
-            .debug(
-                "Found these {} destination header providers for a {}: {}",
-                aggregatedHeaderProviders.size(),
-                getClass().getSimpleName(),
-                Joiner.on(",").join(aggregatedHeaderProviders));
+        final String msg = "Found these {} destination header providers for a {}: {}";
+        log.debug(msg, aggregatedHeaderProviders.size(), getClass().getSimpleName(), aggregatedHeaderProviders);
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java:198]: (minor)
AFAIK log4j can be configured how to log class names.

```suggestion
        log.debug(msg, aggregatedHeaderProviders.size(), getClass(), aggregatedHeaderProviders);
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultServiceBindingDestinationLoaderChain.java [hunk 2]
```diff
@@ -90,13 +87,9 @@ public Try<HttpDestination> tryGetDestination( @Nonnull final ServiceBindingDest
...
-                log
-                    .debug(
-                        "Transformation of service binding ({}) to an {} using an instance of {} {}.",
-                        serviceBindingToString(serviceBinding),
-                        HttpDestination.class,
-                        loader.getClass(),
-                        result.isSuccess() ? "succeeded" : "failed");
+                final String msg = "Transformation of service binding ({}) to an {} using an instance of {} {}.";
+                final String state = result.isSuccess() ? "succeeded" : "failed";
+                log.debug(msg, serviceBindingToString(serviceBinding), HttpDestination.class, loader.getClass(), state);
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultServiceBindingDestinationLoaderChain.java:92]: (medium)

Not sure whether I'm a fan of always executing the `serviceBindingToString` method regardless of whether DEBUG logs are turned on.

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java [hunk 3]
```diff
@@ -52,35 +52,20 @@ public static void setLoader( @Nullable final DestinationLoader loader )
...
             if( loader != null ) {
-                log
-                    .info(
-                        "Setting the current "
-                            + DestinationLoader.class.getSimpleName()
-                            + " used to a custom instance of "
-                            + loader.getClass().getSimpleName()
-                            + ".");
+                final String msg = "Setting the current {} used to a custom instance of {}.";
+                log.info(msg, DestinationLoader.class.getSimpleName(), loader.getClass().getSimpleName());
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java:56]: (minor)

```suggestion
                log.info(msg, DestinationLoader.class, loader.getClass());
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java [hunk 4]
```diff
@@ -52,35 +52,20 @@ public static void setLoader( @Nullable final DestinationLoader loader )
...
             } else {
-                log
-                    .info(
-                        "Resetting the current "
-                            + DestinationLoader.class.getSimpleName()
-                            + " used to the default "
-                            + DestinationLoaderChain.class.getSimpleName()
-                            + ".");
+                final String msg = "Resetting the current {} used to the default {}.";
+                log.info(msg, DestinationLoader.class.getSimpleName(), DestinationLoaderChain.class.getSimpleName());
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java:59]: (minor)

```suggestion
                log.info(msg, DestinationLoader.class, DestinationLoaderChain.class);
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java [hunk 5]
```diff
@@ -52,35 +52,20 @@ public static void setLoader( @Nullable final DestinationLoader loader )
...
     {
-        log
-            .info(
-                "Creating a new DestinationLoaderChain with "
-                    + EnvVarDestinationLoader.class.getSimpleName()
-                    + " as the primary "
-                    + DestinationLoader.class.getSimpleName()
-                    + " implementation.");
+        final String msg = "Creating a new DestinationLoaderChain with {} as the primary {} implementation.";
+        log.info(msg, EnvVarDestinationLoader.class.getSimpleName(), DestinationLoader.class.getSimpleName());
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java:68]: (minor)

```suggestion
        log.info(msg, EnvVarDestinationLoader.class, DestinationLoader.class);
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java [hunk 6]
```diff
@@ -89,13 +74,8 @@ private static DestinationLoader initDestinationLoader()
...
             if( log.isInfoEnabled() ) {
-                log
-                    .info(
-                        "Using an instance of "
-                            + loader.getClass().getSimpleName()
-                            + " as the secondary "
-                            + DestinationLoader.class.getSimpleName()
-                            + " implementation.");
+                final String inf = "Using an instance of {} as the secondary {} implementation.";
+                log.info(inf, loader.getClass().getSimpleName(), DestinationLoader.class.getSimpleName());
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationAccessor.java:78]: (minor)

```suggestion
                log.info(inf, loader.getClass(), DestinationLoader.class);
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java [hunk 7]
```diff
@@ -81,39 +81,26 @@ public static DestinationLoaderChainBuilder builder( @Nonnull final DestinationL
         final ArrayList<Throwable> suppressedList = new ArrayList<>();
 
         for( final DestinationLoader loader : loaders ) {
-            log
-                .debug(
-                    "Delegating destination lookup for destination {} to the destination loader {}.",
-                    destinationName,
-                    loader.getClass().getSimpleName());
+            final String msg = "Delegating destination lookup for destination {} to the destination loader {}.";
+            log.debug(msg, destinationName, loader.getClass().getSimpleName());
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java:85]: (minor)

```suggestion
            log.debug(msg, destinationName, loader.getClass());
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java [hunk 8]
```diff
@@ -81,39 +81,26 @@ public static DestinationLoaderChainBuilder builder( @Nonnull final DestinationL
...
             final Try<Destination> result = loader.tryGetDestination(destinationName, options);
 
             if( result.isSuccess() ) {
-                log
-                    .debug(
-                        "Destination loader {} successfully loaded destination {}.",
-                        loader.getClass().getSimpleName(),
-                        destinationName);
+                final String msg1 = "Destination loader {} successfully loaded destination {}.";
+                log.debug(msg1, loader.getClass().getSimpleName(), destinationName);
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java:91]: (minor)

```suggestion
                log.debug(msg1, loader.getClass(), destinationName);
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java [hunk 9]
```diff
@@ -81,39 +81,26 @@ public static DestinationLoaderChainBuilder builder( @Nonnull final DestinationL
...
 
             if( !hasCauseAssignableFrom(cause, DestinationNotFoundException.class) ) {
-                log
-                    .debug(
-                        "Destination loader {} returned an exception when loading destination {}:",
-                        loader.getClass().getSimpleName(),
-                        destinationName,
-                        cause);
+                final String msg1 = "Destination loader {} returned an exception when loading destination {}:";
+                log.debug(msg1, loader.getClass().getSimpleName(), destinationName, cause);
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java:99]: (minor)

```suggestion
                log.debug(msg1, loader.getClass(), destinationName, cause);
```

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java [hunk 10]
```diff
@@ -81,39 +81,26 @@ public static DestinationLoaderChainBuilder builder( @Nonnull final DestinationL
...
                 return result;
             }
-            log
-                .debug(
-                    "No destination with name '{}' was found in destination loader {}.",
-                    destinationName,
-                    loader.getClass().getSimpleName());
+            final String msg1 = "No destination with name '{}' was found in destination loader {}.";
+            log.debug(msg1, destinationName, loader.getClass().getSimpleName());
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationLoaderChain.java:103]: (minor)

```suggestion
            log.debug(msg1, destinationName, loader.getClass());
```

File: cloudplatform/servlet-jakarta/src/main/java/com/sap/cloud/sdk/cloudplatform/servletjakarta/RequestAccessorFilter.java [hunk 11]
```diff
@@ -67,13 +67,8 @@ public void doFilter(
...
             if( log.isWarnEnabled() ) {
-                log
-                    .warn(
-                        "Failed to initialize "
-                            + ThreadContext.class.getSimpleName()
-                            + ": request not of type "
-                            + HttpServletRequest.class.getName()
-                            + ".");
+                final String msg = "Failed to initialize {}: request not of type {}.";
+                log.warn(msg, ThreadContext.class.getSimpleName(), HttpServletRequest.class.getName());
```
Comment (Johannes-Schneider) [inline cloudplatform/servlet-jakarta/src/main/java/com/sap/cloud/sdk/cloudplatform/servletjakarta/RequestAccessorFilter.java:71]: (minor)

```suggestion
                log.warn(msg, ThreadContext.class, HttpServletRequest.class);
```

File: cloudplatform/servlet/src/main/java/com/sap/cloud/sdk/cloudplatform/servlet/RequestAccessorFilter.java [hunk 12]
```diff
@@ -83,13 +83,8 @@ public void doFilter(
...
             if( log.isWarnEnabled() ) {
-                log
-                    .warn(
-                        "Failed to initialize "
-                            + ThreadContext.class.getSimpleName()
-                            + ": request not of type "
-                            + HttpServletRequest.class.getName()
-                            + ".");
+                final String msg = "Failed to initialize {}: request not of type {}.";
+                log.warn(msg, ThreadContext.class.getSimpleName(), HttpServletRequest.class.getName());
```
Comment (Johannes-Schneider) [inline cloudplatform/servlet/src/main/java/com/sap/cloud/sdk/cloudplatform/servlet/RequestAccessorFilter.java:87]: (minor)

```suggestion
                log.warn(msg, ThreadContext.class, HttpServletRequest.class);
```

**Review by Johannes-Schneider — COMMENTED** [review]
Generally LGTM - just one mediocre comment regarding the execution of `serviceBindingToString` method.

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java (hunk 1 above)
Comment (newtork) [reply cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java:198]: I went down the `SimpleLogger` implementation of **SLF4J** and couldn't find any special handling of `Class` references.

<details>
<summary>Looks like this.</summary>

image

</details>

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultServiceBindingDestinationLoaderChain.java (hunk 2 above)
Comment (newtork) [reply cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultServiceBindingDestinationLoaderChain.java:92]: Please find the `isDebugEnabled()` if-condition above the quoted code.

**Review by Johannes-Schneider — APPROVED** [review]
LGTM

**Pull request merged**